### ISSUE TITLE
WiX 3.6 Requires .NET 2.0

### ISSUE THREAD

**Issue by wixbot** [wixtoolset/issues#3872]
This was tested with v3.6 (sorry, didn't test 3.7 since I stick with final releases).  Basically I was setting up a minimalistic build server for TFS Service.  I installed Windows 8, Team Foundation Server and WiX 3.6.  This left me with _ONLY_ .NET 4.5.   My first build threw the following error:

C:\Program Files (x86)\MSBuild\Microsoft\WiX\v3.x\wix2010.targets (2339): The "WixAssignCulture" task could not be loaded from the assembly C:\Program Files (x86)\WiX Toolset v3.6\bin\WixTasks.dll. Could not load file or assembly 'Microsoft.Build.Utilities, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.

I was able to add the .NET 3.5 feature and green up the build.  It seems like this should work with only .NET 4.5 installed _OR_ the WiX installer should detect a missing prereq.

> Originally opened by `chrpai` from [http://sourceforge.net/p/wix/bugs/3175/](http://sourceforge.net/p/wix/bugs/3175/)

**Comment by rajasekarshanmugam** [comment]
WixTasks.dll from version (WiX Toolset v3.10.2) also has a reference to Microsoft.Build.Utilities, Version=2.0.0.0, Microsoft.Build.Framework, Version=3.5.0.0. Got the same error on a new Win10/VS2015+Update 1 setup with only 4.6+ framework installed. Hope this info is of some help. I am wondering if any *.config, assembly redirections will help. Last choice is to install the older version of .NET.

**Comment by rajasekarshanmugam** [comment]
Found a quick/dirty workaround to get past the above error (WixTasks.dll) and following steps were followed:
- Close VS
- Copy wixtasks.dll from official build to a temp folder, keep a backup copy of the original.
- Open VS developer command prompt, cd to temp folder created above - run command - ildasm /out=WixTasks.il WixTasks.dll
- Open WixTasks.il using a good text editor. Replace - .assembly extern 'Microsoft.Build.Utilities' with .assembly extern 'Microsoft.Build.Utilities.v4.0' at the very top of the file. Find and replace [Microsoft.Build.Utilities'] with ['Microsoft.Build.Utilities.v4.0'] (approx 209 replacements)
- Download key from https://github.com/wixtoolset/wix3/blob/develop/tools/publickey_wix.snk
- Run command - ilasm /dll /key:publickey_wix.snk /output=WixTasks.dll WixTasks.il
- Replace the official copy with the new WixTasks.dll (typically found at an equivalent location 'C:\Program Files (x86)\WiX Toolset v3.10\bin' unless default directory was changed)
- Try MSBuild or VS, should at least get past of only WixTasks.dll was bothering

Hope it helps somebody, until an official solution arrives.

**Comment by robmen** [comment]
@rajasekarshanmugam WiX v3.x still supports MSBuild v3.5 thus the NETFX v3.5 build dependency. MSBuild v3.5 support is cut in WiX v4.0.

**Comment by rajasekarshanmugam** [comment]
@robmen, Apologize for the late response, but yes, will try the v4.0 builds.

**Comment by ahaid** [comment]
I tried with this [4.0 build](http://wixtoolset.org/releases/v4-0-3922-0/) on a fresh Windows 10 machine with Visual Studio 2015 Update 2. The dependency on .NET Framework 3.5 is still present. I was unable to compile my wix project until I turned on the 3.5 framework feature. After enabling it and changing the schema of my .wxs file to `<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">`, it compiled fine.
